packages/node/src/methods/controller.ts

```diff
@@ -4,35 +4,21 @@ import { Controller } from "rpc-server";
 
 import { RequestHandler } from "../request-handler";
 
+import { executeFunctionWithinQueues } from "./queued-execution";
+
 export abstract class NodeController extends Controller {
   public static readonly methodName: Node.MethodName;
 
   public async executeMethod(
     requestHandler: RequestHandler,
     params: Node.MethodParams
   ): Promise<Node.MethodResult> {
-    const shardedQueues = await this.enqueueByShard(requestHandler, params);
-
-    let promise;
-
-    const executeCached = async () => {
-      if (!promise) {
-        promise = this.executeMethodImplementation(requestHandler, params);
-      }
-      return await promise;
-    };
-
     await this.beforeExecution(requestHandler, params);
 
-    let ret;
-
-    if (shardedQueues.length > 0) {
-      for (const queue of shardedQueues) queue.add(executeCached);
-      for (const queue of shardedQueues) await queue;
-      ret = await promise;
-    } else {
-      ret = await executeCached();
-    }
+    const ret = await executeFunctionWithinQueues(
```

Comment:
Better variable name.

Reply:
There actually isn't a better name IMO, it's literally an `any` type response

Reply:
Wasn't referring to the type, just the name. Not a blocker.